workshops/hcload/README.md

```diff
@@ -0,0 +1,465 @@
+
+1. Start a file server on our device
+2. Expose the file server to the internet using [ngrok](https://ngrok.com/)
+3. Send the URL of the ngrok instance to the API
+
+![Visual diagram of the process above: 1. A request is sent from the program to the CDN with the array of URLs. 2. The server sends a request to the ngrok instance as per the URL provided. 3. The ngrok instance responds with the file contents 4. The server hosts the files and responds with the hosted file URLs](https://cloud-hrkhc4qna.vercel.app/0cdn-workflow.png)
+
+## Part 3: Programming the library
+### Part 3.1: Setting up the server
+
+Open up `mod.ts`, and let's get started! 
+
+- First, let's create our HTTP server. We'll be using [Oak](https://github.com/oakserver/oak), a server middleware framework for Deno.
+
+```js
+import { Application, send } from 'https://deno.land/x/oak@v6.3.1/mod.ts'
+
+export default async function (): Promise<void> {
+        const app = new Application()
+        app.use(async (context: any) => {
+            context.response.body = "Server running"
+        })
+
+        app.addEventListener("listen", async ({ port }) => {
+            console.log("HTTP server ready")
+        })
+
+        await app.listen({ port: 20685 })
+}
+```
+
+At the moment, the server simply replies with "Server running" when anything is requested. If you trying running the program (simply run hcload), you should see HTTP server ready, and on the top left of your screen "Server running"
+
+- Next, let's ask for a filePath in our library's default function, and serve the file when an HTTP request reaches our server
+
+```js
+import { Application, send } from 'https://deno.land/x/oak@v6.3.1/mod.ts'
+
+export default async function (filePath: string): Promise<void> { // <--- Changed
+    const app = new Application()
+    app.use(async (context: any) => {
+        await send(context, filePath, { root: '/' }) // <--- Changed
+    })
+
+    app.addEventListener("listen", async ({ port }) => {
+        console.log("HTTP server ready")
+    })
+
+    await app.listen({ port: 20685 })
+}
+```
+
+Modify `hcload.ts` to pass a filePath to the library. Currently, you can pass the `LICENSE` file included with the starter code.
+
+```js
+import * as path from "https://deno.land/std@0.75.0/path/mod.ts"
+import hcload from "./mod.ts"
+
+hcload(path.resolve("./LICENSE"))
+```
+
+Run `hcload` and you should see the contents of the file `LICENSE` in your browser preview. Congratulations on making it so far!
+
+### Part 3.2: Exposing localhost via ngrok
+
+At the moment, files served will only be accessible over your home network (yes, over repl.it they'll be accessible over the internet, but the end use of this program is to upload files from your own computer). Since the Hackclub CDN needs to be able to make a request to your server over the internet, we'll need to expose the server over the internet.
```

Comment:
Wait I'm a little confused. The explanation in the parentheses doesn't make sense to me. I guess it'll make more sense to me when I reach the end of the workshop, but I think this should be a little clearer